client/components/Solution.js

```diff
@@ -1,63 +1,74 @@
-import React, { useState, useEffect } from 'react';
+import React, { Component } from 'react';
 import CodeMirror from 'react-codemirror';
 import { connect } from 'react-redux';
 import { fetchOneChallenge, fetchUserchallengeById } from '../store';
 import { convertBufferToImgSrc } from '../utils';
 
-// two props passed down from route parameters
-// challengeId and userchallengeId
-const _Solution = ({
-  user,
-  individualChallenge,
-  userchallenge,
-  challengeId,
-  userchallengeId,
-  fetchOneChallenge,
-  fetchUserchallengeById,
-}) => {
-  useEffect(() => {
-    if (challengeId) {
-      fetchOneChallenge(challengeId);
-    }
-    if (userchallengeId) {
-      fetchUserchallengeById(userchallengeId);
-    }
-  }, []);
+// todo convert to functional component
+class _Solution extends Component {
+  constructor() {
+    super();
+    this.state = {
+      userchallenge: {},
+    };
+  }
 
-  const options = {
-    lineNumbers: true,
-    mode: 'javascript',
-  };
-  if (!individualChallenge.solutions) {
-    return null;
+  componentDidMount() {
+    this.props.fetchOneChallenge(this.props.challengeId);
+    this.props.fetchUserchallengeById(this.props.userchallengeId);
```

Comment:
Do you want to set this on state since you did set this on state on componentDidUpdate?